### PR TITLE
feat: fixes setting of authentication realm url.

### PR DESCRIPTION
**Pull request by marcosuma**
## Description

When authenticating to App Service (Realm) there is a bug that impedes using a `base_url` different than `realm.mongodb.com`. As such, using `realm-dev` is not possible. With this fix, we are able to successfully authenticate. This is not an external bug.

Link to any related issue(s): https://jira.mongodb.org/browse/INTMDB-1010

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [X] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [X] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [X] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code
- [X] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/mongodb/terraform-provider-mongodbatlas
 
-go 1.20
+go 1.21
```
Comment (lantoli) [inline go.mod]: I'm not sure we should upgrade to Go 1.21 until we decide so, i don't know that is the process or how time we normally wait until we consider a version stable.
and in any case I think this is important enough to be in their own PR, and also other files must be changed like .tools-versions

File: go.sum [hunk 2]
```diff
@@ -186,6 +187,7 @@ cloud.google.com/go/webrisk v1.5.0/go.mod h1:iPG6fr52Tv7sGk0H6qUFzmL3HHZev1htXuW
 cloud.google.com/go/workflows v1.6.0/go.mod h1:6t9F5h/unJz41YqfBmqSASJSXccBLtD1Vwf+KmJENM0=
 cloud.google.com/go/workflows v1.7.0/go.mod h1:JhSrZuVZWuiDfKEFxU0/F1PQjmpnpcoISEXH2bcHC3M=
 dario.cat/mergo v1.0.0 h1:AGCNq9Evsj31mOgNPcLyXc+4PNABt905YmuqPYYpBWk=
+dario.cat/mergo v1.0.0/go.mod h1:uNxQE+84aUszobStD9th8a29P2fMDhsBdgRYvZOxGmk=
```
Comment (lantoli) [inline go.sum]: i suppose all these changes are related to go version change?

File: mongodbatlas/config.go [hunk 3]
```diff
@@ -97,10 +98,12 @@ func (c *MongoDBClient) GetRealmClient(ctx context.Context) (*realm.Client, erro
 	}
 
 	optsRealm := []realm.ClientOpt{realm.SetUserAgent(userAgent)}
+	authConfig := realmAuth.NewConfig(nil)
 	if c.Config.BaseURL != "" && c.Config.RealmBaseURL != "" {
 		optsRealm = append(optsRealm, realm.SetBaseURL(c.Config.RealmBaseURL))
+		authConfig.AuthURL, _ = url.Parse(c.Config.RealmBaseURL + "api/admin/v3.0/auth/providers/mongodb-cloud/login")
 	}
-	authConfig := realmAuth.NewConfig(nil)
+
```
Comment (lantoli) [inline mongodbatlas/config.go:106]: is it ok if you only change this file to fix the issue?

Comment (marcosuma) [reply mongodbatlas/config.go:106]: Yes :)

File: go.sum (hunk 2 above)
Comment (marcosuma) [reply go.sum]: thanks for flagging this @lantoli - removed these changes as the problem was as @AgustinBettati  said (I was using 1.21 on my IDE).

**Review by lantoli — APPROVED** [review]
